include/swift/Serialization/ModuleFormat.h

```diff
@@ -1545,12 +1545,11 @@ namespace decls_block {
   using AvailableDeclAttrLayout = BCRecordLayout<
     Available_DECL_ATTR,
     BCFixed<1>, // implicit flag
-    BCFixed<1>, // is unconditionally unavailable?
-    BCFixed<1>, // is unconditionally deprecated?
     BC_AVAIL_TUPLE, // Introduced
     BC_AVAIL_TUPLE, // Deprecated
     BC_AVAIL_TUPLE, // Obsoleted
     BCVBR<5>,   // platform
+    BCVBR<5>,   // platform agnostic kind
```

Comment:
I am not sure how appropriate it is to change the serialization format. I could add another boolean but storing the enum seemed much cleaner.

Reply:
Storing the enum isn't great because now people have to remember to update the module format number when they add a new case to the enum. That's why all the serialization routines bounce through a "raw" version of the enum that mirrors all the cases. You could either do that or go back to booleans.